controller/server/gui.ml

```diff
@@ -263,57 +261,125 @@ module NetworkGui = struct
       >|= List.filter (fun s -> not internet_connected || s |> Service.is_connected)
     in
 
+    let blur_service_proxy_password s =
+      let open Service in
+      let blur p = Proxy.validate p |> Option.map (Proxy.to_string ~hide_password:true) in
+      { s with proxy = s.proxy |> Base.Fn.flip Option.bind blur }
+    in
+
+    let after_restart_proxy =
+      services
+      |> List.find_map (fun s ->
+          if s.Service.autoconnect then
+            Option.bind s.Service.proxy Proxy.validate
+          else
+            None)
+    in
+
+    let proxy_infos =
+      [ proxy |> Option.map (Proxy.to_string ~hide_password:true)
+      ; (if after_restart_proxy <> proxy then
+          (match after_restart_proxy with
+          | Some p -> Some ("after restart: " ^ (Proxy.to_string ~hide_password:true p))
+          | None -> Some "after restart: removed")
+        else
+          None)
+      ]
+      |> Base.List.concat_map ~f:Option.to_list
+    in
```

Comment:
This permits to give useful insights to the user, indicating that his configuration is saved, but will only be applied after a restart.

Reply:
No longer showed.